frame/rewards/src/lib.rs

```diff
@@ -101,10 +120,20 @@ decl_module! {
 					reward = T::Reward::get();
 				}
 
-				drop(pallet_balances::Module::<T>::deposit_creating(&author, reward));
+				let tax = Self::taxation() * reward;
+				let donate = min(
+					tax,
+					Self::author_donation().map(|dp| dp * reward).unwrap_or(T::Balance::zero())
+				);
+
+				let miner = reward.saturating_sub(tax);
+
+				drop(pallet_balances::Module::<T>::deposit_creating(&author, miner));
+				drop(pallet_balances::Module::<T>::deposit_creating(&treasury_id, donate));
```

Comment:
Seems like we should have some check on both the reward and donation amount that they are above the Existential Deposit.